src/templates/seo-head.js

```diff
@@ -30,22 +30,46 @@ export const nuxtI18nSeo = function () {
   const link = []
   // hreflang tags
   if (strategy !== STRATEGIES.NO_PREFIX) {
-    link.push(...this.$i18n.locales
-      .map(locale => {
-        if (locale[LOCALE_ISO_KEY]) {
-          return {
-            hid: 'alternate-hreflang-' + locale[LOCALE_ISO_KEY],
-            rel: 'alternate',
-            href: baseUrl + this.switchLocalePath(locale.code),
-            hreflang: locale[LOCALE_ISO_KEY]
-          }
-        } else {
-          // eslint-disable-next-line no-console
-          console.warn(`[${MODULE_NAME}] Locale ISO code is required to generate alternate link`)
-          return null
-        }
-      })
-      .filter(item => !!item))
+    const languageFromLocaleIso = iso => iso.split('-')[0] // en-US -> en
+
+    const locales = this.$i18n.locales.map(locale => {
+      const localeIso = locale[LOCALE_ISO_KEY]
+      if (!localeIso) {
+        // eslint-disable-next-line no-console
+        console.warn(`[${MODULE_NAME}] Locale ISO code is required to generate alternate link`)
+        return
+      }
+
+      return {
+        hid: `alternate-hreflang-${localeIso}`,
+        rel: 'alternate',
+        href: baseUrl + this.switchLocalePath(locale.code),
+        hreflang: localeIso
+      }
+    }).filter(Boolean)
+
+    link.push(...locales)
+
+    const languages = locales.map(locale => languageFromLocaleIso(locale.hreflang))
+    const uniqueLanguages = Array.from(new Set(languages))
+
+    const catchAllLocales = uniqueLanguages.map(language => {
+      const localesWithLanguage = this.$i18n.locales.filter(locale => languageFromLocaleIso(locale[LOCALE_ISO_KEY]) === language)
+      if (!localesWithLanguage) {
+        return
+      }
+
+      const forcedCatchallLocale = localesWithLanguage.find(locale => locale.isCatchallLocale) || localesWithLanguage[0]
+      const catchAllLocaleIso = forcedCatchallLocale[LOCALE_ISO_KEY]
+      const catchAllLanguage = languageFromLocaleIso(catchAllLocaleIso)
+      const catchallLocale = { ...locales.find(locale => locale.hreflang === catchAllLocaleIso) }
+      catchallLocale.hid = `alternate-hreflang-${catchAllLanguage}`
+      catchallLocale.hreflang = catchAllLanguage
+
+      return catchallLocale
+    }).filter(Boolean)
```

Comment:
Find this code kinda hard to read with all the maps, filters, finds and multiple uses of `languageFromLocaleIso`.

I've come up with this instead which I think works as well (at least tests think so):
```js
  if (strategy !== STRATEGIES.NO_PREFIX) {
    const localeMap = new Map()

    for (const locale of this.$i18n.locales) {
      const localeIso = locale[LOCALE_ISO_KEY]

      if (!localeIso) {
        // eslint-disable-next-line no-console
        console.warn(`[${MODULE_NAME}] Locale ISO code is required to generate alternate link`)
        continue
      }

      const [language, region] = localeIso.split('-')

      if (language && region && (locale.isCatchallLocale || !localeMap.has(language))) {
        localeMap.set(language, locale)
      }

      localeMap.set(localeIso, locale)
    }

    for (const [iso, locale] of localeMap.entries()) {
      link.push({
        hid: `alternate-hreflang-${iso}`,
        rel: 'alternate',
        href: baseUrl + this.switchLocalePath(locale.code),
        hreflang: iso
      })
    }
  }
```

Also, we could think about moving this generation to the module's code as it needs to be done only once really. The `switchLocalePath` would still have to be done at runtime though.

Reply:
Really like this code and added it together with my other refactoring in the latest commits.

Regarding extracting the majority of the setup (where possible) to the build time part - this would totally make sense, as long as the complexity of the runtime part with `switchLocalePath` won't be increased ☺️

Reply:
We should also consider that `Map` might need a polyfill, so moving that part to runtime could also remove the need for that ☺️